acmed/config/nginx_hooks.toml

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) 2021 Rodolphe Bréard <rodolphe@breard.tf>
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# ------------------------------------------------------------------------
+# ACMEd hooks adapting nginx configuration
+# You should not edit this file since it may be overridden by a newer one.
+# ------------------------------------------------------------------------
+
+
+###
+# nginx file storing challenge root in "/etc/nginx/conf.d/{{challenge-location}}/"
+# env: NGINX_CONFDIR -> /etc/nginx/conf.d
+# env: NGINX_CHALLENGE_LOCTION -> 001-challenge-letsencrypt.conf
+###
+
+[[hook]]
+name = "nginx-config-challenge-location-chmod"
+type = ["challenge-http-01", "post-operation"]
+cmd = "chmod"
+args = [
+    "ug+rw",
+    "{{#if env.NGINX_CONFDIR}}{{env.NGINX_CONFDIR}}{{else}}/etc/nginx/conf.d{{/if}}/{{#if env.NGINX_CHALLENGE_LOCATION}}{{env.NGINX_CHALLENGE_LOCATION}}{{else}}001-challenge-letsencrypt.conf{{/if}}"
+]
+allow_failure = true
+
+[[hook]]
+name = "nginx-config-challenge-location-create"
+type = ["challenge-http-01", "post-operation"]
+cmd = "cat"
+args = [ "-" ]
+stdin_str = """###
+# Let's Encrypt: acme-challenge location
+###
+
+location ^~ /.well-known/acme-challenge/ {
+  allow all;
+  root "{{#if env.HTTP_ROOT}}{{env.HTTP_ROOT}}{{else}}/var/lib/acmed/domains{{/if}}";
+  default_type "text/plain";
+  try_files $uri =404;
+}
+"""
+stdout = "{{#if env.NGINX_CONFDIR}}{{env.NGINX_CONFDIR}}{{else}}/etc/nginx/conf.d{{/if}}/{{#if env.NGINX_CHALLENGE_LOCATION}}{{env.NGINX_CHALLENGE_LOCATION}}{{else}}001-challenge-letsencrypt.conf{{/if}}"
+
+[[hook]]
+name = "nginx-config-challenge-location-echo"
+type = ["challenge-http-01", "post-operation"]
+cmd = "echo"
+args = [
+    "{{#if env.NGINX_CONFDIR}}{{env.NGINX_CONFDIR}}{{else}}/etc/nginx/conf.d{{/if}}/{{#if env.NGINX_CHALLENGE_LOCATION}}{{env.NGINX_CHALLENGE_LOCATION}}{{else}}001-challenge-letsencrypt.conf{{/if}}"
+]
+allow_failure = true
+
+[[group]]
+name = "nginx-config-challenge-location"
```

Comment:
Maybe the name should include `http-01` so people would know it's only about this challenge.

Reply:
Isn't this just about the location for `.well-known`? Does any other challange variant vary in its requests? The name string is already a long one. Will adapt, it is realy gains any advantages.
`dns-01` doesn't need it. How about `tls-alpn-01` ?

Reply:
`tls-alpn-01` doesn't use `.well-known` either, this challenge rely on a an ALPN extension to TLS and is therefore resolved during the TLS handshake. No file is served by the server. You can have a look at tacd which is the recommended tool to solve this challenge with ACMEd.

As you guessed, `http-01` is the only challenge to use it, that why I would recommend to add it in the name of hooks that relies on it.